### PR TITLE
feat: improve native app feel

### PR DESCRIPTION
**Pull request by jamesbrink**
## Linked Issue

No linked issue — UX-polish branch curated as a single coherent improvement to overall app feel. Happy to break into per-area issues if preferred before review.

## Summary

Five tightly related improvements to how the app *feels* on macOS:

1. **Native window materials & vibrancy** — translucent panes that read as macOS-native rather than a flat WebView. Commits: `feat: enable native window materials`, `feat: make frontend panes feel more native`, `fix: enable macos private api feature`.

2. **Title-bar polish** — the workspace branch label drags the window without selecting text; clicks on header action buttons no longer accidentally drag; app-icon thumbnails render with neutral backgrounds instead of bright white. Commits: `fix: keep header branch label draggable`, `fix: prevent header branch text selection`, `fix: separate header drag and action regions`, `fix: keep app icon backgrounds neutral`, `fix: start header label window drag`.

3. **No more chat rubber-band** — the chat panel no longer elastically bounces past the top/bottom on macOS WebView. Scroll chaining from nested scrollables (code blocks, thinking blocks, attachment previews, slash-command picker) into the chat scroller works correctly. Commits: `fix: prevent chat scroll rubber band`, `fix: harden chat overscroll guard`, plus the global `* { overscroll-behavior: contain }` removed in `5462d161` after Codex flagged it as breaking scroll chaining.

4. **Scrollbars unified across chat / Monaco / xterm** — same 8px width, themed thumb color (`var(--scrollbar-thumb)`), 4px rounded corners (`var(--radius-sm)`), and 8px right-edge gap on all three surfaces. They scale together with browser zoom (previously the chat used the macOS OS overlay, which doesn't scale, so the three diverged at non-100% zoom). Implementation: a small `OverlayScrollbar.tsx` custom DOM scrollbar component mirroring xterm.js's pattern (track + slider, ResizeObserver + MutationObserver + scroll listener, RAF-coalesced updates, pointer-capture drag, track click-to-page). Monaco's slider was rounded via a `:global` rule (Monaco's theme schema has no slider-radius token).

5. **Click-to-page restored on chat scrollbar** — clicking the chat scrollbar track above or below the slider pages by a viewport-height. Matches native macOS scrollbar + Monaco. Was previously absent because the chat used the OS overlay only.

## Screenshots

Visual changes are subtle (translucent material, scrollbar geometry parity); best appreciated live rather than in screenshots. Will attach a side-by-side scrollbar comparison if requested.

## UAT Plan

- [ ] Resize chat past the viewport — slider appears, sized proportional to viewport/content, draggable
- [ ] Click chat scrollbar track above / below the slider — viewport pages by ~its own height in that direction
- [ ] At 100% zoom, chat / Monaco / xterm sliders look identical (same width, color, radius, right-edge gap)
- [ ] `Cmd+`+`/`-` to zoom the WebView — all three sliders scale together
- [ ] Hover slider — tint changes to `--scrollbar-thumb-hover`
- [ ] Stream a long agent reply — slider updates smoothly, no per-token re-render lag
- [ ] Scroll chat to top / keep scrolling — no rubber-band bounce
- [ ] Scroll a nested code block at its top — wheel chains into the chat scroller (regression check for the global `* { overscroll-behavior: contain }` removal)
- [ ] Drag the workspace branch label — window moves, no text selection
- [ ] Click any header action button — no window drag fires
- [ ] Open Settings / Files / Diff Viewer panels — they still scroll (note: those still use OS overlay scrollbars; extending parity to them is a follow-up — see below)

## Checklist

- [x] CI passes — `bunx tsc -b`, `bun run lint` (0 errors), `bun run lint:css`, `bun run test` (1694/1694)
- [x] New or updated tests cover the changed behaviour — `OverlayScrollbar.test.tsx` (20 tests: render, overflow gating, geometry math, scroll/resize/mutation reactivity, RAF coalescing, drag mapping/clamping, track click-to-page, edge cases, cleanup); `scrollbarConsistency.test.ts` (12 tests: cross-file pin that chat / Monaco / xterm CSS modules all reference the shared `--scrollbar-*` / `--radius-sm` tokens)
- [ ] Documentation updated if public behaviour changed — **intentional N/A**: changes are pure visual/interaction polish, no settings / CLI flags / persisted state introduced. Per CLAUDE.md "Documentation discipline" section, calling out the omission deliberately.
- [x] PR title follows [conventional commit](../CONTRIBUTING.md#commit-conventions) format

## Peer-review notes

Two reviewers (`codex review --base main` and a Claude general-purpose subagent) ran on the branch:

- **Codex** flagged one P2: the global `* { overscroll-behavior: contain }` rule on `theme.css:892` regressed wheel-scroll chaining for nested scrollable widgets and undermined `usePreventScrollBounce`. Addressed in `5462d161` (rule removed; `html, body, #root { overscroll-behavior: none }` and the dedicated bounce-prevention hook are sufficient).
- **Claude subagent** flagged: streaming-token MutationObserver thrash → RAF-coalesced via `scheduleUpdate()` matching `useStickyScroll`'s pattern; missing track click-to-page → added; missing edge-case tests for null `targetRef` and `usableTrack === 0` drag → added. All addressed in `5462d161`.

## Follow-up (out of scope for this PR)

The same WKWebView zoom-drift bug we just fixed for chat / Monaco / xterm exists on every other scrollable surface in the app — sidebar workspace list, right sidebar (files/changes/tasks), file tree, file viewer, diff viewer, settings, command palette, fuzzy finder, modals, several chat sub-pickers (slash-command, file-mention, thinking blocks, attachment previews), dashboard. Extending parity to those surfaces is a separate piece of work — likely lifting `OverlayScrollbar` out of `chat/` into a shared component plus a `useOverlayScrollbar(ref)` hook so other consumers can adopt it without re-implementing positioning. Worth tracking as its own issue.

### CODE REVIEW

File: src/ui/src/components/chat/OverlayScrollbar.tsx [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+    const minHeight = 24;
+    const rawHeight = (clientHeight / scrollHeight) * clientHeight;
+    const height = Math.max(minHeight, rawHeight);
+    const maxTop = clientHeight - height;
+    // `scrollHeight - clientHeight > 0` is guaranteed by the `overflow`
+    // gate above — the divide-by-zero branch never executes here.
+    const scrollProgress = scrollTop / (scrollHeight - clientHeight);
+    const top = Math.round(scrollProgress * maxTop);
+    setSliderHeight(height);
+    setSliderTop(top);
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/components/chat/OverlayScrollbar.tsx:90]: Addressed in `ddf6dc82`. Clamped slider height to `min(clientHeight, max(minHeight, rawHeight))` and `maxTop` to `max(0, ...)` so the geometry stays valid when the user resizes the chat panel below 24px. Added an explanatory comment to the math; the existing `usableTrack === 0` drag test in `OverlayScrollbar.test.tsx` already covers the degenerate-rail case.

File: src/ui/src/hooks/usePreventScrollBounce.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+/**
+ * Prevents WebKit's elastic overscroll on desktop webviews. CSS
+ * `overscroll-behavior` is not reliable enough in macOS WKWebView, so this
+ * cancels wheel/touch gestures only when the active scroll surface is already
+ * at the top or bottom edge.
+ */
+export function usePreventScrollBounce(
+  containerRef: RefObject<HTMLElement | null>,
+) {
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/hooks/usePreventScrollBounce.ts:100]: Addressed in `ddf6dc82`. Exported the four pure boundary-detection helpers (`canScrollVertically`, `canScrollInDirection`, `nearestScrollableWithin`, `boundaryScrollTarget`) and added `usePreventScrollBounce.test.tsx` with 22 cases — the predicate matrix (overflow gating with +1px subpixel tolerance, top/bottom edge direction blocking, scroll chaining via nested scrollables, `overflowY: visible` disqualification) plus three end-to-end tests that mount the hook and assert `preventDefault` fires when (and only when) the gesture would bounce off the boundary. The "nested scroller can chain" case explicitly pins the regression that motivated removing the global `* { overscroll-behavior: contain }` rule earlier in this branch.

File: src/ui/src/components/chat/scrollbarConsistency.test.ts [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+// Update notes (read before editing): if you intentionally change the
+// token strategy (e.g., introducing a separate `--chat-scrollbar-size`),
+// update both the surface AND this test in the same commit, and call
+// out the divergence in the PR description so reviewers know it was
+// deliberate.
+
+import { readFileSync } from "node:fs";
+import { join } from "node:path";
+import { describe, expect, it } from "vitest";
+
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/components/chat/scrollbarConsistency.test.ts]: Addressed in `ddf6dc82`. Switched to `fileURLToPath(import.meta.url)` + `dirname(...)` to match the pattern in `components/layout/headerAlignment.test.ts`. Test no longer depends on Vitest's `__dirname` shim and is portable to bare Node.
